apps/meteor/server/models/raw/Rooms.ts

```diff
@@ -1327,9 +1330,12 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 				},
 				includeFederatedRooms
 					? {
-							$or: [{ $and: [{ $or: [{ federated: { $exists: false } }, { federated: false }], name }] }, { federated: true, fname: name }],
+							$or: [
```

Comment:
If you're adding nameCondition to both conditions, why not moving it to the main query?

Reply:
Not sure if I can do that due to [line 1335](https://github.com/RocketChat/Rocket.Chat/pull/29804/files#diff-4812f4201fc467c5333be9eba5448d71fa67a6ad0e0be89e9645dddcae5066ccR1335)
When `federated` is true, we should only search by `fname` :thinking: I think that's intended behavior